### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Malahov-Alexandr**
Created a program that adds publication to a file

### CODE REVIEW

File: OOP/News.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# 1. Пользователь выбирает, какой тип данных он хочет добавить
+# 2. Предоставьте требуемые данные типа записи
+# 3.Запись публикуется в текстовом файле в специальном формате
+
+# Вам необходимо реализовать:
+# 1.Новости – текст и город в качестве ввода. Дата рассчитывается во время публикации.
+# 2.Приватное объявление – на входе текст и дата истечения срока действия. Оставшийся день рассчитывается во время публикации.
+# 3. Ваш уникальный с уникальными правилами публикации.
+
+# Created a constractor witch will add new publication to the txt file
```
Comment (arastopchin1) [inline OOP/News.py]: Like 'Ведьма'? :)

File: OOP/News.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# Создайте инструмент, который будет создавать новостную ленту, сгенерированную пользователем:
+# 1. Пользователь выбирает, какой тип данных он хочет добавить
+# 2. Предоставьте требуемые данные типа записи
+# 3.Запись публикуется в текстовом файле в специальном формате
+
+# Вам необходимо реализовать:
+# 1.Новости – текст и город в качестве ввода. Дата рассчитывается во время публикации.
+# 2.Приватное объявление – на входе текст и дата истечения срока действия. Оставшийся день рассчитывается во время публикации.
+# 3. Ваш уникальный с уникальными правилами публикации.
+
```
Comment (arastopchin1) [inline OOP/News.py]: Please no Russian in the code, task is in English, why did you translate it?

File: OOP/News.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+# Created a constractor witch will add new publication to the txt file
+class File_creator:
+
+    def record_to_file(self, text):
+        with open("OOP/result.txt", "a+") as f:
+            f.write(text)
+
+# create a class wich will be creating a news
```
Comment (arastopchin1) [inline OOP/News.py]: Not even 'Ведьма' here))

File: OOP/News.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+class News(File_creator):
```
Comment (arastopchin1) [inline OOP/News.py]: Definitely here should not be Inheritance, FileCreator class logically is not a parent of News or Adv. It makes no sense.

File: OOP/News.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+# 2. Предоставьте требуемые данные типа записи
+# 3.Запись публикуется в текстовом файле в специальном формате
+
+# Вам необходимо реализовать:
+# 1.Новости – текст и город в качестве ввода. Дата рассчитывается во время публикации.
+# 2.Приватное объявление – на входе текст и дата истечения срока действия. Оставшийся день рассчитывается во время публикации.
+# 3. Ваш уникальный с уникальными правилами публикации.
+
+# Created a constractor witch will add new publication to the txt file
+class File_creator:
```
Comment (arastopchin1) [inline OOP/News.py]: Please check PEP8 about naming conventions of classes in Python

File: OOP/News.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+    def record_to_file(self, text):
```
Comment (arastopchin1) [inline OOP/News.py]: PyCharm IDE even said that it is a static class. Please check what it is and add special decorator.

File: OOP/News.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        with open("OOP/result.txt", "a+") as f:
+            f.write(text)
+
+# create a class wich will be creating a news
+class News(File_creator):
+
+    def __init__(self, city, text):
+        self.city = city
+        self.text = text
+    # created constructer wich will be creating a news and add it to the txt file
```
Comment (arastopchin1) [inline OOP/News.py]: Please, follow English syntax everywhere. What is 'constracter'?

File: OOP/News.py [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+    def new_constr(self):
+        import datetime
```
Comment (arastopchin1) [inline OOP/News.py]: I already mentioned before that all imports are in the very beginning of a file.

File: OOP/News.py [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+        text = f'\nNews {"-" * 25}\n{self.text}\n{self.city}, {datetime.date.today()}\n{"-" * 30}\n\n'
```
Comment (arastopchin1) [inline OOP/News.py]: What are 25 and 30 magic numbers? Avoid magic numbers in your code. Read about this antipattern.

Comment (arastopchin1) [reply OOP/News.py]: And please make your code more granular. No need to put everything in one code line, it is hard to read and understand.

File: OOP/News.py [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+        gener = File_creator()
```
Comment (arastopchin1) [inline OOP/News.py]: Please no shorten names without clear reason for it. Why for instance gener but not 'file_generator', 'text_generator', 'news_generator' etc.

File: OOP/News.py [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+        text = f'\nPrivate Ad{"-" * 20}\n{self.text}\n{self.calculate_date()}, {self.date} days left\n\n '
+        gener = File_creator()
+        gener.record_to_file(text)
+
+
+class Joke(File_creator):
+    def __init__(self, text, answer, rate):
+        self.text = text
+        self.rate = rate
+        self.answer = answer
```
Comment (arastopchin1) [inline OOP/News.py]: It also would be great if you added docstrings which could explain what exactly this class about, which parameters it takes, and talking about functions and methods - which arguments they get and what this function does.

File: OOP/News.py [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+            elif text == 'joke':
+                text = input('Please, enter the joke question: ')
+                answer = input('Please, enter the answer for the joke: ')
+                rate = int(input('Enter the rate of this joke: '))
+                myobj = Joke(text, answer, rate)
+                myobj.joke_generator()
+            elif text == 'stop':
+                break
+            else:
+                print('Incorrect enter. Please, only News,Adv or 0')
```
Comment (arastopchin1) [inline OOP/News.py]: Incorrect input, what is 0?

File: OOP/News.py [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+class Generator(Adv, News):
+
+    @classmethod
+    def comunnication_with_user(cls):
+        while True:
+            text = input('Please, input the type of publication (News or Adv or Joke) if you do not want to add a publication enter "stop": ').lower()
+            if text == 'news':
+                text = input('Please, enter your text of the news: ')
+                city = input('Please, enter the city where it happend: ')
+                myobj = News(city, text)
```
Comment (arastopchin1) [inline OOP/News.py]: No myobj. Name it 'news' for example, may be 'news_article'; myobj explain nothing.

File: OOP/News.py [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+        self.answer = answer
+
+    def joke_generator(self):
+        text = f'\nJoke{"-" * 26}\n{self.text}\n{self.answer}\nFunny meter  {self.rate} of 10\n\n'
+        gener = File_creator()
+        gener.record_to_file(text) 
+
+
+# created class wich invoke a class after wheh user will be asked about a type of publication
+class Generator(Adv, News):
```
Comment (arastopchin1) [inline OOP/News.py]: First you create inheritance of news and adv from TextCreator then you inherit Generator from them? I can not see a big sense in it. Moreover why have you decided that 'communication_with_user' where you again made a syntax mistake is a class method? May be static, may be even just function not a class, but not a class method definitely.

**Review by arastopchin1 — CHANGES_REQUESTED** [review]

File: OOP/News.py (hunk 1 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 2 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 3 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 4 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 5 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 6 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 7 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 8 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 9 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 10 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 11 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 12 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 13 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: fixed

File: OOP/News.py (hunk 14 above)
Comment (Malahov-Alexandr) [reply OOP/News.py]: Fixed

File: OOP/News.py [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+# created constructor which will be creating an adv and add it to the txt file
+class Adv:
+    def __init__(self, text, date):
+        # the text of the adv - str
+        self.text = text
+        # date after how many days adv will be expired- int
+        self.date = date
+
+    def calculate_date(self):
+        import datetime
```
Comment (arastopchin1) [inline OOP/News.py:45]: ?

File: OOP/News.py [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+
+import datetime
+
+
+def record_to_file(text):
+    with open("result.txt", "a+") as f:
+        f.write(text)
+
+
+# create a class which will be creating a news
```
Comment (arastopchin1) [inline OOP/News.py:20]: https://www.geeksforgeeks.org/python-docstrings/

**Pull request merged**